src/server/game/Miscellaneous/SharedDefines.h

```diff
@@ -2687,6 +2687,11 @@ inline constexpr uint64 IMMUNE_TO_MOVEMENT_IMPAIRMENT_AND_LOSS_CONTROL_MASK(\
     (1 << MECHANIC_TURN) | (1 << MECHANIC_HORROR) | (1 << MECHANIC_DAZE) | \
     (1 << MECHANIC_SAPPED));
 
+inline constexpr uint64 MECHANIC_LOSS_CONTROL_MASK(\
+    (1 << MECHANIC_CHARM) | (1 << MECHANIC_FEAR) | (1 << MECHANIC_STUN) | (1 << MECHANIC_KNOCKOUT) | \
+    (1 << MECHANIC_SLEEP) | (1 << MECHANIC_DISORIENTED) | (1 << MECHANIC_POLYMORPH) | \
+    (1 << MECHANIC_BANISH) | (1 << MECHANIC_HORROR));
```

Comment:
should root, shackle, sap, freeze (is this even used nowdays?) also be taken into account?

Reply:
i get the info in wowpedia so could be incomplete xd